### ISSUE TITLE
DOC: Line chart with point markers

### ISSUE THREAD

**Issue by eitanlees** [altair-viz/altair#876]
I thought a nice adition to the documentation would be a simple line chart with point markers.

As an example let's use the [simple line chart](https://altair-viz.github.io/gallery/simple_line_chart.html) from the example gallery. 

```python
import altair as alt
import numpy as np
import pandas as pd

x = np.arange(100)
data = pd.DataFrame({'x': x,
                     'sin(x)': np.sin(x / 5)})
```

I found out there are two ways to do this. 

- Create the line and points seperately, and layer

```python
base = alt.Chart(data).encode(
    x='x',
    y='sin(x)'
)

base.mark_line() + base.mark_circle()
```
![layer](https://user-images.githubusercontent.com/2041969/40310112-c5562d82-5cd9-11e8-8cf4-d89f8a3b5041.png)

- Create the line with `point=True` as a parameter. 

```python
alt.Chart(data).mark_line(point=True).encode(
    x='x',
    y='sin(x)'
)
```
![pointstrue](https://user-images.githubusercontent.com/2041969/40310150-d807a280-5cd9-11e8-9b32-41246dd185d4.png)

I am not sure which is prefered. 

There are a few other examples with points and lines where changes could be made such as

- [Connected Scatterplot (Lines with Custom Paths)](https://altair-viz.github.io/gallery/connected_scatterplot.html)

- [Multiple Marks](https://altair-viz.github.io/gallery/multiple_marks.html)

**Comment by jakevdp** [comment]
This would be a nice example!

**Comment by jakevdp** [comment]
I think either approach is fine.

**Comment by eitanlees** [comment]
Closed by #877
